artifactory/commands/transferconfig/transferconfigmerge.go

```diff
@@ -0,0 +1,373 @@
+	if s == "" {
+		s = property
+		return s
+	}
+	s = s + ", " + property
+	return s
+}
+
+func (tcc *TransferConfigCommand) tryPing(targetServicesManager artifactory.ArtifactoryServicesManager) error {
+	_, err := targetServicesManager.Ping()
+	return err
+}
+
+func (tcc *TransferConfigCommand) validateMergeArtifactoryServers(targetServicesManager artifactory.ArtifactoryServicesManager, sourceArtifactoryVersion string, minRequiredVersion string) (bool, error) {
+	// if version is less than 7.0.0 projects are not defined, and we don't have to transfer them
+	transferProjects := true
+	if !version.NewVersion(sourceArtifactoryVersion).AtLeast(minRequiredVersion) {
+		transferProjects = false
+	}
+
+	// Avoid exporting and importing to the same server
+	log.Info("Verifying source and target servers are different...")
+	if tcc.sourceServerDetails.GetArtifactoryUrl() == tcc.targetServerDetails.GetArtifactoryUrl() {
+		return false, errorutils.CheckErrorf("The source and target Artifactory servers are identical, but should be different.")
+	}
+
+	// check correctness of Authorization
+	log.Info("Checking validation of your authorization methods..")
+	if tcc.tryPing(targetServicesManager) != nil {
+		return false, errorutils.CheckErrorf("The target's access token is not correct, please provide an availble access token.")
+	}
+	return transferProjects, nil
+}
+
+// Create a csv summary of all conflicts
+func (tcc *TransferConfigCommand) createConflictsCSVSummary(conflicts []ProjectConflict, timeStarted time.Time) (csvPath string, err error) {
+	// Create CSV file
+	summaryCsv, err := loguitils.CreateCustomLogFile(fmt.Sprintf("transfer-config-conflicts-%s.csv", timeStarted.Format(loguitils.DefaultLogTimeLayout)))
+	if err != nil {
+		return
+	}
+	csvPath = summaryCsv.Name()
+	defer func() {
+		e := summaryCsv.Close()
+		if err == nil {
+			err = e
+		}
+	}()
+	// Marshal JSON typed FileWithLongProperty array to CSV file
+	err = errorutils.CheckError(gocsv.MarshalFile(conflicts, summaryCsv))
+	return
+}
+
+func (tcc *TransferConfigCommand) mergeRepositories(sourceServiceManager, targetServiceManager artifactory.ArtifactoryServicesManager) (err error) {
+	sourceRepos, err := sourceServiceManager.GetAllRepositories()
+	if err != nil {
+		return
+	}
+	targetRepos, err := targetServiceManager.GetAllRepositories()
+	if err != nil {
+		return
+	}
+	targetReposMap := make(map[string]services.RepositoryDetails)
+	for _, repo := range *targetRepos {
+		targetReposMap[repo.Key] = repo
+	}
+	reposToTransfer := []string{}
+	var repoConflicts []RepositoryConflict
+	for _, sourceRepo := range *sourceRepos {
+		// Check if repository is filtered
+		var shouldIncludeRepo bool
+		shouldIncludeRepo, err = tcc.getRepoFilter().ShouldIncludeRepository(sourceRepo.Key)
+		if err != nil {
+			return
+		}
+		if !shouldIncludeRepo {
+			continue
+		}
+
+		if targetRepo, exists := targetReposMap[sourceRepo.Key]; exists {
+			// Repository exists on target, need to compare
```

Comment:
// Repository exists on target, need to compare
-->
// The repository exists on target. We need to compare the repositories.